### ISSUE TITLE
When uploading specific video to Twitter, processing always fails.

### ISSUE THREAD

**Issue by GodsWithin** [linvi/tweetinvi#1026]
I have tried numerous times to upload this video: `https://scontent-bru2-1.cdninstagram.com/v/t50.2886-16/91339123_105462207681438_3583484007519639729_n.mp4?_nc_ht=scontent-bru2-1.cdninstagram.com&_nc_cat=106&_nc_ohc=zJQ3ItflSWIAX_q0Xq7&oe=5E8275E8&oh=457512edc20a9d7ab7fc22503d8c4687`

Code:
```cs
HttpClient http = new HttpClient();
byte[] bytes = await http.GetByteArrayAsync(...);
IMedia videoMedia = Upload.UploadVideo(bytes);
Upload.WaitForMediaToBecomeAvailable(videoMedia);
```

Uploading always succeeds, but processing always fails, thus when I try to publish a tweet with this media object, it will not publish and the returned tweet will be `null`.

Any idea if this is a library issue or a Twitter issue?

**Comment by linvi** [comment]
Hello,

I will try to upload the video on my side and check the errors received.
Video formats on twitter are pretty limited on the public api and this could be why Twitter is not processing your image.

I am not sure I will find time to check today but I will try.

Cheers,
Linvi

**Comment by GodsWithin** [comment]
@linvi 

I played around with it some more and came to the conclusion that, as the error message said that the video's GOP was to large, greater than 512. Twitter doesn't like this so it probably invalidates the mediaId that the upload returned.

I ended up having to use ffmpeg to re-encode the videofile and specifying in the options that the maximum GOP is to be 512. After this re-encoding I was able to tweet out the video.
